src/app/bookshelf/addToBookshelfPanel/useAddBookToBookshelf.tsx

```diff
@@ -0,0 +1,52 @@
+"use client";
+import { useState, useEffect } from "react";
+import { useAddBookToBookshelfMutation } from "@/api/bookshelfApi";
+import { fetchBookByIsbn } from "@/api/fetchBookByIsbn";
+import { AddBookshelfBook } from "@/interfaces/BookshelfBook";
+import { useTranslations } from "next-intl";
+
+interface Status {
+  status: "success" | "error";
+  message: string;
+}
+
+export function useAddBookToBookshelf() {
+  const t = useTranslations();
+  const [status, setStatus] = useState<Status | undefined>(undefined);
+  const [addBookToBookshelf, { isLoading, isSuccess, isError }] =
+    useAddBookToBookshelfMutation();
+
+  useEffect(() => {
```

Comment:
`useEffect` usually introduces another re-render cycle. This looks like a code that can be constructed using only `useMemo`. Also, I'd move the messages (`i18n`) to be closer to rendering and not translate them here, but in the display component, based on the internal error state.